src/openloco/windows/build_vehicle.cpp

```diff
@@ -86,40 +455,954 @@ namespace openloco::ui::build_vehicle
+                found = true;
+                break;
+            }
+        }
+
+        window->current_secondary_tab = found ? trackTab : 0;
+
+        bool isRoad = _TrackTypesForTab[trackTab] & (1 << 7);
+        uint8_t trackType = _TrackTypesForTab[trackTab] & ~(1 << 7);
+        setTopToolbarLastTrack(trackType, isRoad);
+    }
+
+    // 0x4A3A06
+    static void setTopToolbarLastTrack(uint8_t trackType, bool isRoad)
+    {
+        bool setRail = false;
+        if (isRoad)
+        {
+            auto road_obj = objectmgr::get<road_object>(trackType);
+            if (road_obj && road_obj->flags & flags_12::unk_01)
+            {
+                setRail = true;
+            }
+        }
+        else
+        {
+            auto rail_obj = objectmgr::get<track_object>(trackType);
+            if (rail_obj && !(rail_obj->flags & flags_22::unk_02))
+            {
+                setRail = true;
+            }
+        }
+
+        if (setRail)
+        {
+            last_railroad_option = trackType;
+        }
+        else
+        {
+            last_road_option = trackType;
+        }
+
+        // The window number doesn't really matter as there is only one top toolbar
+        WindowManager::invalidate(WindowType::topToolbar, 0);
+    }
+
+    // 0x4C2865 common for build vehicle window and vehicle list
+    static void setTransportTypeTabs(ui::window* window)
+    {
+        auto disabledWidgets = window->disabled_widgets >> widx::tab_build_new_trains;
+        auto widget = window->widgets + widx::tab_build_new_trains;
+        auto tabWidth = widget->right - widget->left;
+        auto tabX = widget->left;
+        for (auto i = 0; i <= widx::tab_build_new_ships - widx::tab_build_new_trains; ++i, ++widget)
+        {
+            if (disabledWidgets & (1ULL << i))
+            {
+                widget->type = widget_type::none;
+            }
+            else
+            {
+                widget->type = widget_type::wt_8;
+                widget->left = tabX;
+                widget->right = tabX + tabWidth;
+                tabX += tabWidth + 1;
+            }
+        }
+    }
+
+    /* 0x4B60CC
+     * Opens vehicle window and clicks???
+     */
+    static void sub_4B60CC(openloco::vehicle* vehicle)
+    {
+        registers regs;
+        regs.edx = (int32_t)vehicle;
+        call(0x4B60CC, regs);
+    }
+
+    // 0x4C2BFD
+    static void drawTransportTypeTabs(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        auto skin = objectmgr::get<interface_skin_object>();
+        auto companyColour = companymgr::get_company_colour(window->number);
+
+        for (auto tab : _transportTypeTabInformation)
+        {
+            auto frameNo = 0;
+            if (_transportTypeTabInformation[window->current_tab].type == tab.type)
+            {
+                frameNo = (window->frame_no / 2) & 0xF;
+            }
+            uint32_t image = gfx::recolour(skin->img + tab.imageId + frameNo, companyColour);
```

Comment:
While this is likely the most efficient manner, for the other windows we defined a lookup table for all animation frames. The compiler is able to optimise this, and it leaves room for the images to not be in sequence.

I don't feel too strongly about this, but @marijnvdwerf asked me to change mine to lookup tables in the past. I'll leave it to you.

Reply:
Yes i see that now looking through the other works. Will change it.